### ISSUE TITLE
pycodcif parser: more documentation is needed

### ISSUE THREAD

**Issue by blokhin** [cod-developers/cod-tools#4]
Dear Andrius, Saulius and the COD development team,

thanks for `pycodcif`, I enjoy it!

Please let me share some minor issues which I have encountered while playing with it. First, pypi mentions `http://wiki.crystallography.net/cod-tools` as a homepage, however no single word about `pycodcif` is there. Second, what are the `options` provided for `parse`? Why `CifParserException` is unused? Finally, if I provide a string instead of CIF filename, a very irrelevant stacktrace is generated.

**Comment by merkys** [comment]
Dear Evgeny,

thank you for the report. Indeed, the documentation of the package is close to none (as of now). I will list the options in wiki ASAP.

Meanwhile I have done some testing with non-existing file names to elicit these two situations:
* non-existing ``str`` filename:
```
from pycodcif import parse
parse( "test" )
```
Output (trimmed for brevity):
```
/usr/bin/ipython: test: ERROR, could not open file -- no such file or directory.
...
CifParserException: /usr/bin/ipython: test: ERROR, could not open file -- no such file or directory.
```
This is expected behavior: ``CifParserException`` is thrown and the message is appropriate here.

* non-existing ``unicode`` filename:
```
from pycodcif import parse
parse( u"test" )
```
Output (trimmed for brevity):
```
TypeError: in method 'parse_cif', argument 1 of type 'char *'
```
This is clearly a bug. Is this the one that you have encountered?

**Comment by merkys** [comment]
@blokhin, the proper homepage for ``pycodcif`` is [http://wiki.crystallography.net/cod-tools/CIF-parser/](http://wiki.crystallography.net/cod-tools/CIF-parser/), I have updated the package description.

**Comment by merkys** [comment]
I have fixed the issue with ``unicode`` filenames in ``pycodcif`` Python package version 0.8.7.

**Comment by blokhin** [comment]
@merkys thanks 👍

**Comment by blokhin** [comment]
I meant the whole CIF contents can be in a string. _e.g._ 'Pymatgen' has 'from_string' method for that.

**Comment by merkys** [comment]
This is not supported in ``pycodcif`` yet. To get around you may use ``NamedTemporaryFile`` from [tempfile](https://docs.python.org/2/library/tempfile.html).

**Comment by blokhin** [comment]
Yep. I'm asking because CIF in a string is a typical usecase within the client-server architecture (relevant for me)

**Comment by merkys** [comment]
I have put this in a wishlist ([http://bugz.crystallography.net/bugzilla3/show_bug.cgi?id=127](http://bugz.crystallography.net/bugzilla3/show_bug.cgi?id=127)).

**Comment by blokhin** [comment]
@merkys http://bugz.crystallography.net does not work anymore :cry:

**Comment by blokhin** [comment]
@merkys it seems I'm not the only one suffering from `pycodcif`'s lack of support of reading CIF from a string. ASE has exactly the same problem:

```
from io import StringIO
from ase.io import read

buffer = StringIO(cif_content_in_a_string)
parsed_cif = read(buffer, format='cif', reader='pycodcif')
```

This ends up with the buffer misuse error.

**Comment by merkys** [comment]
> @merkys http://bugz.crystallography.net does not work anymore :cry:

We have moved to another issue tracker: https://projects.ibt.lt/repositories/projects/cod-tools/issues (`pycodcif` issue seems to be lost in transition...)

`pycodcif` does not support reading CIF from a string yet. I recall working to that direction, but I dropped the ball at some point. Now I have better understanding of C/Python interface, thus I should revisit the topic.
